covin_slot_tracker.py

```diff
@@ -61,52 +67,56 @@ def checkAvailability(payload, age):
                 for j in range(sessions_len):
-                    if((payload['centers'][i]['sessions'][j]['available_capacity']>0) and
-                       (payload['centers'][i]['sessions'][j]['min_age_limit']<=age)):
+                    if((payload['centers'][i]['sessions'][j]['available_capacity'] > 0) and
+                       (payload['centers'][i]['sessions'][j]['min_age_limit'] <= age)):
                         available_centers.add(payload['centers'][i]['name'])
-            available_centers_str =  ", ".join(available_centers)
+            available_centers_str = ", ".join(available_centers)
             total_available_centers = len(available_centers)
-    
-    return available_centers_str,total_available_centers
+
+    return available_centers_str, total_available_centers
 
 
-if __name__=="__main__":
+if __name__ == "__main__":
     with open("settings.json") as f:
         settings = json.load(f)
 
     # Load from JSON file
-    key_list = ["districtId","authToken","accountSID","twilioPhone","selfPhone","userAge"]
+    key_list = ["districtId", "authToken", "accountSID",
+                "twilioPhone", "selfPhone", "userAge"]
     if all(key in settings for key in key_list):
         DISTRICT_ID = settings["districtId"]
         SECRET_TOKEN = settings["authToken"]
         ACCOUNT_SID = settings["accountSID"]
         TWILIO_PHONE_NUMBER = settings["twilioPhone"]
         CELL_PHONE_NUMBER = settings["selfPhone"]
         USER_AGE = int(settings["userAge"])
-        
+
         client = Client(ACCOUNT_SID, SECRET_TOKEN)
         while(True):
-            date = getDate()
-            data1 = pingCOWIN(date,DISTRICT_ID)
-            available, total_centers = checkAvailability(data1,USER_AGE)
-            if available:
-                msg_body = "Slots Available at {total} places.\n{available}".format(total = total_centers,available = available)
-                print(msg_body)
-                client.messages.create(from_=TWILIO_PHONE_NUMBER,
-                           to=CELL_PHONE_NUMBER,
-                           body= msg_body)
-            else:
-                print("No Available Centers")
-            
+            sms_sent = False
+            for i in range(90):
+                date = getDate(i)
+                data1 = pingCOWIN(date, DISTRICT_ID)
+                available, total_centers = checkAvailability(data1, USER_AGE)
+                if available:
+                    msg_body = "Slots available for {date} at {total} places.\n{available}".format(
+                        date=date, total=total_centers, available=available)
+                    print(msg_body)
+                    if not sms_sent:
+                        client.messages.create(from_=TWILIO_PHONE_NUMBER,
+                                               to=CELL_PHONE_NUMBER,
+                                               body=msg_body)
+                        sms_sent = True
```

Comment:
If sms_sent is set to True, then it makes sense to come out of the loop

Reply:
No, sms_sent is set to true after sms is sent for current WHILE ITERATION, Even if 80 out of 90 days are returning as available it only sends sms for first day remaining all it just prints.
 After time.sleep(900) it again goes for next iteration of while loop and set send_sms to true.

Reply:
Why do we need 3 months advance notification? This whole project is supposed to notify immediate slots.

Reply:
because when I checked on cowin it is booked for weeks already and checking for current day only makes so sense when even next days are also booked.